aws-xray-agent/src/main/java/com/amazonaws/xray/agent/runtime/models/XRayTransactionState.java

```diff
@@ -100,4 +105,8 @@ public static void setServiceName(String inServiceName) {
     public static String getServiceName() {
         return serviceName;
     }
+
+    public static Map<PreparedStatement, String> getPreparedStatementMap() {
```

Comment:
Is there any concurrency concern?

Reply:
Good callout, forgot to address this. Unfortunately there is no WeakConcurrentHashMap in the JDK, so going to use this library for a weak concurrent hashmap: https://github.com/raphw/weak-lock-free

It is currently used by the OTel Java library for similar use cases (though they extend it to add other functionality).